sliderule_dsi_xml_exercise_branch.ipynb

```diff
@@ -0,0 +1,1486 @@
+    "print \"Longest river: \" + river_name\n",
+    "print \"Length: \" + str(river_size)\n",
+    "print \"Country of source: \" + river_country"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Largest laker: Caspian Sea\n",
+      "Size: 386400.0\n",
+      "Countries passed: ['Russia', 'Azerbaijan', 'Kazakhstan', 'Iran', 'Turkmenistan']\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Search data set for largest lake\n",
+    "# This search bases country on list of all countries passed by river\n",
+    "lake_size=0.0\n",
+    "lake_name=[]\n",
+    "lake_country=[]\n",
+    "for child in document.getroot().iterfind('lake[area]'):\n",
+    "    if float(child.find('.area').text) > lake_size:\n",
+    "        lake_size = float(child.find('.area').text)\n",
+    "        lake_name = child.find('.name').text\n",
+    "        lake_country = list(map(lambda x: country_code[x], child.attrib[\"country\"].split()))\n",
+    "print \"Largest laker: \" + lake_name\n",
+    "print \"Size: \" + str(lake_size)\n",
+    "print \"Countries passed: \" + str(lake_country)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Hightest elevation airport: El Alto Intl\n",
+      "Elevation: 4063.0\n",
+      "Country: ['Bolivia']\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Search data set for highest elevalation airport\n",
+    "airport_height=0.0\n",
+    "airport_name=[]\n",
+    "airport_country=[]\n",
+    "for child in document.getroot().iterfind('airport[elevation]'):\n",
+    "    if (child.find('.elevation').text is not None) and (float(child.find('.elevation').text) > airport_height):\n",
+    "        airport_height = float(child.find('.elevation').text)\n",
+    "        airport_name = child.find('.name').text\n",
+    "        airport_country = list(map(lambda x: country_code[x], child.attrib[\"country\"].split()))\n",
+    "        \n",
+    "print \"Hightest elevation airport: \" + airport_name\n",
+    "print \"Elevation: \" + str(airport_height)\n",
+    "print \"Country: \" + str(airport_country)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# Created repeatable code to perfom searches based on solution for airport\n",
+    "def search_xml(s_item, s_feat):\n",
+    "    airport_height=0.0\n",
+    "    airport_name=[]\n",
+    "    airport_country=[]\n",
+    "    for child in document.getroot().iterfind((s_item)+'['+s_feat+']'):\n",
+    "        if (child.find('.'+s_feat).text is not None) and (float(child.find('.'+s_feat).text) > airport_height):\n",
```

Comment:
Use spaces for more legible code and adhering to the PEP8 style guide 
https://www.python.org/dev/peps/pep-0008/
http://docs.python-guide.org/en/latest/writing/style/
Tools like https://pypi.python.org/pypi/autopep8 can help you automatically format your code